src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1502,7 +1502,13 @@ def aks_create(cmd, client, resource_group_name, name, ssh_key_value,  # pylint:
                            'Are you an Owner on this subscription?')
 
     network_profile = None
-    if any([network_plugin, pod_cidr, service_cidr, dns_service_ip, docker_bridge_address, network_policy]):
+    if any([pod_cidr, service_cidr, dns_service_ip, docker_bridge_address, network_policy]):
+        if not network_plugin:
+            logger.warning('Please explicitly specify network plugin type')
+            return
+        if pod_cidr and network_plugin == "azure":
+            logger.warning('Please use kubenet as a network plugin when pod_cidr is specified')
+            return
```

Comment:
Seems to me like these should be usage errors, not warnings, but I will defer to AKS folks on whatever they prefer.

Reply:
You're right. I will change them to raise a CLIError.